packages/framework/ini-files/config-specs.ini

```diff
@@ -2854,7 +2854,7 @@ opt-set-cmake-var Teko_DISABLE_LSCSTABALIZED_TPETRA_ALPAH_INV_D          BOOL
 opt-set-cmake-var KokkosKernels_blas_serial_MPI_1_DISABLE                BOOL         : ON
 opt-set-cmake-var ROL_example_PDE-OPT_helmholtz_example_02_MPI_1_DISABLE BOOL         : ON
 opt-set-cmake-var ROL_example_PDE-OPT_navier-stokes_example_01_MPI_4_DISABLE BOOL     : ON
-opt-set-cmake-var CMAKE_CXX_FLAGS                                        STRING FORCE : ${CMAKE_CXX_FLAGS|CMAKE} -Wall -Wno-clobbered -Wno-vla -Wno-pragmas -Wno-unknown-pragmas -Wno-unused-local-typedefs -Wno-literal-suffix -Wno-deprecated-declarations -Wno-misleading-indentation -Wno-int-in-bool-context -Wno-maybe-uninitialized -Wno-nonnull-compare -Wno-address -Wno-inline -DTRILINOS_HIDE_DEPRECATED_HEADER_WARNINGS
```

Comment:
It's the other way around.  `${CMAKE_CXX_FLAGS|CMAKE}` here will work for generating cmake fragments but will fail in the shell.  
```
ValueError: Unresolved variable expansion for `CMAKE_CXX_FLAGS` in a BASH file. CMake variables are only valid in a CMake fragment file.
```
The first time `CMAK_CXX_FLAGS` is set it cannot reference itself or else you get the error above.  With it removed, it works in both cases.  Not having it is also consistent with all but 1 of the other build specs.

Reply:
So you're saying don't inherit any previously-set values from this variable now, correct?  I'm concerned that will change behavior.  Do any of the previous `use` directives set this variable?

Reply:
No nothing sets it before this which is the problem.  GenConfig does not like it when you reference a variable that is not set.  So the first time `CMAKE_CXX_FLAGS` is being set is on this line.   It also works to set it to nothing before any `use` statements.  so this works:

```
[rhel8_gcc-openmpi_debug_shared_no-kokkos-arch_no-asan_complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables]
#uses a containerized environment                                                                                                                                         
opt-set-cmake-var CMAKE_CXX_FLAGS STRING FORCE :
use COMPILER|GNU
use NODE-TYPE|SERIAL
use BUILD-TYPE|DEBUG

...

opt-set-cmake-var CMAKE_CXX_FLAGS                                        STRING FORCE : ${CMAKE_CXX_FLAGS|CMAKE} -Wall -Wno-clobbered -Wno-vla -Wno-pragmas -Wno-unknown-\
pragmas -Wno-unused-local-typedefs -Wno-literal-suffix -Wno-deprecated-declarations -Wno-misleading-indentation -Wno-int-in-bool-context -Wno-maybe-uninitialized -Wno-no\
nnull-compare -Wno-address -Wno-inline -DTRILINOS_HIDE_DEPRECATED_HEADER_WARNINGS
```
Then it will get any changes from subsequent `use`'s and will be set if nothing else sets it.  What do you think of that approach?

Reply:
Now that you've explained it I'm fine with either, I leave it to your discretion.